src/appendix/git.md

```diff
@@ -0,0 +1,199 @@
+details how to setup git for your system.
+
+### Linux
+
+On a Debian based Linux distribution you can install `git` with the following
+command:
+
+```sh
+$ sudo apt-get install git
+```
+
+### Mac OS X
+
+There is a binary installer for `git` that can be found at
+[http://git-scm.com/download/mac](http://git-scm.com/download/mac).
+
+### Windows
+
+I hope to provide better instructions for windows soon. If you know how consider
+[contributing][] to the book.
+
+## Starting
+
+git can keep track of changes to your project. To get started with tracking your
+project you need to initialize git. In the root of your directory execute the
+command
+
+```sh
+$ git init
+```
+
+This asks the git to start a repository that will track changes to your project.
+
+## Status
+
+With the previous command you created a repository. But git is not tracking anything
+yet! Issue the next command to see what git knows about your project.
+
+```sh
+$ git status
+```
+
+This will give the following output.
+
+```text
+On branch master
+
+Initial commit
+
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+
+	Makefile
+	boot.asm
+	build/
+	grub.cfg
+	linker.ld
+	multiboot_header.asm
+
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+One important section is the `untracked files` section. Here git lists the files
+that are not tracked yet. You can recognize the files that you created for
+intermezzOS.
+
+## Adding
+
+If you want git to keep track of changes to your project you need to tell which
+files you are interested in. You can do this with the `add` command. For example
+the following command adds the `Makefile`
+
+```sh
+$ git add Makefile
+```
+
+If you issue the `git status` command again you see a change in the output.
+
+```text
+On branch master
+
+Initial commit
+
+Changes to be committed:
+  (use "git rm --cached <file>..." to unstage)
+
+	new file:   Makefile
+
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+
+	boot.asm
+	build/
+	grub.cfg
+	linker.ld
+	multiboot_header.asm
+
+```
+
+We will use the `git add` command again to add the other files that we want git
+to track.
+
+```sh
+$ git add boot.asm grub.cfg linker.ld multiboot_header.asm
+```
+
+Note that we left out `build/` directory. We will get back to that in a moment.
+
+## Commiting
+
+We added files to git's staging area. The staging area is used to create a nice
+commit. A commit is like a save point. Ideally you want a commit to be something
+you can go back to safely.
+
+Once you are happy with the files in the staging area you can commit them with
+the following command.
+
+```sh
+$ git commit
+```
+
+An editor will open so you can provide a _commit message_, a description of the
+commit that will help in identifying what changed. For example,
+
+```text
+A minimal OS that outputs "Hello, World!"
+```
+
+## Ignoring
```

Comment:
I wonder if this should be the first thing people do before learning to add files. Impatient people like me would likely have already added and commited files :scream: